packages/p2p/src/components/modal-manager/modals/leave-page-modal.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { Button, Modal, Text } from '@deriv/components';
+import { observer } from 'mobx-react-lite';
+import { useStores } from 'Stores';
+import { Localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+
+const LeavePageModal = ({ onLeavePage = () => {} }) => {
+    const { buy_sell_store } = useStores();
+    const { hideModal, is_modal_open } = useModalManagerContext();
+
+    return (
+        <Modal
+            has_close_icon={false}
+            is_open={is_modal_open}
+            small
+            title={
+                <Text color='prominent' size='s' weight='bold'>
+                    <Localize i18n_default_text='Leave page?' />
+                </Text>
+            }
+        >
+            <Modal.Body>
+                <Text color='prominent' size='xs'>
+                    <Localize i18n_default_text='Are you sure you want to leave this page? Changes made will not be saved.' />
+                </Text>
+            </Modal.Body>
+            <Modal.Footer>
+                <Button
+                    large
+                    onClick={() => {
```

Comment:
could we maybe create a function for onClick and then pass it to onClick?

`const onClick() => buy_sell_store.setShowFilterPaymentMethods(true);
                hideModal({
                       should_restore_local_state: true,
                 });
`

```suggestion
                    const onClick{onClick}
```